TU20Bot/Configuration/EmailChecker.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Discord;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TU20Bot.Configuration {
+    public class EmailChecker {
+
+        private Config config;
+        private Client client;
+        private DbCommUnverifiedUser dbComm;
+
+        public EmailChecker(Config config, Client client, DbCommUnverifiedUser dbComm) {
+            this.config = config;
+            this.client = client;
+            this.dbComm = dbComm;
+        }
+
+
+        // Method running on a separate thread and mathcing any unverified email to the email list in csv
+        public async Task emailCheck(List<CSVData> csvEmail) {
+
+            var userInfo = checkEmailInCsvList(csvEmail);
+
+            if (userInfo.userData != null) {
+                await assignRole(userInfo);
+                Console.WriteLine($"user email with id: {userInfo.userId} verified");
+            }
+
+        }
+
+        // Method comparing and returing the CSVData and ulong of the user who's email have been verified
+        public (ulong userId, CSVData userData) checkEmailInCsvList(List<CSVData> csvData) {
+
+            var unverifiedUserList = dbComm.getUserList();
+
+            foreach (var unverifiedUser in unverifiedUserList) {
+
+                // Comparing all emails in the newly obtained csv list with unverified emails
+                foreach (var botUser in csvData) {
+
+                    // If some unverified email matches the email from the csv list,
+                    if (botUser.Email.Equals(unverifiedUser.Email)) {
+
+                        // Get the user id of user associated with that email
+                        ulong Id = unverifiedUser.UserId;
+
+                        // Remove that specific index from the dictionary since the user has been verified
+                        dbComm.removeUserInfo(unverifiedUser);
+
+                        return (userId: Id, userData: botUser);
+                    }
+                }
+            }
+
+            return (userId: ulong.MinValue, userData: null);
+        }
+
+
+        public async Task assignRole((ulong userId, CSVData userData) userInfo) {
```

Comment:
```suggestion
        public async Task assignRole(ulong userId, CSVData userData) {
```

tuplesepction whaduheck